### ISSUE TITLE
[java] LinguisticNaming flags Predicates with boolean-style names

### ISSUE THREAD

**Issue by SamCarlberg** [pmd/pmd#1362]
**Affects PMD Version:**  
6.7.0

**Rule:**  
[LinguisticNaming](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#linguisticnaming)

**Description:**  
Predicates can be stored in fields or local variables to reduce lambda or method reference allocations, and often use boolean-style naming prefixes ("is", "has", etc.) like a boolean-returning method. Methods with this naming scheme are fine, but predicates (and relations such as `BiPredicate`) trigger the rule.

It may be best to add a property to allow custom predicate-like types to pass the rule as well, since there may be interfaces in use like this:
```java
interface Filter<T> {
  boolean passes(T value);
}
```

**Code Sample demonstrating the issue:**

```java
private final Predicate<String> isEmpty = String::isEmpty;
```

**Running PMD through:**  
Gradle
